src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Reload/ReloadableProject.cs

```diff
@@ -143,7 +153,13 @@ public Task Initialize()
                         // The only safe thing to do at this point is to reload the project in the solution
                         // TODO: should we have an additional return value here to indicate that the existing project could be in a bad
                         // state and the reload needs to happen without the user being able to block it?
-                        _telemetryService.Report("projectreload/reload-failed", "Project Reload Failure due to general fault", ex);
+                        var correlationEvent = _telemetryService.Report(ReloadFailedNFWPath, "Project Reload Failure due to general fault", ex);
+                        _telemetryService.PostOperation(
```

Comment:
if you only care about failure case, why not just make all those NFW? NFW won't create dump on every single call. it will be throttled so user won't get affected as much. but we have dump for all failed case so to find out what caused those. and it automatically save data in kusto so you can search every single case. 

if you care both success and failed case, now you can calculate failure rate. and you can create simple IDisposable pattern that calls StartOperation and EndOperation and dispose and put whole body in using.